sofar/sofar.py

```diff
@@ -0,0 +1,2125 @@
+            keys_b = [k for k in keys_b if not k.startswith("GLOBAL_")]
+        elif exclude.upper() == "ATTR":
+            keys_a = [k for k in keys_a if
+                      sofa_a._convention[k]["type"] != "attribute"]
+            keys_b = [k for k in keys_b if
+                      sofa_b._convention[k]["type"] != "attribute"]
+        elif exclude.upper() == "DATE":
+            keys_a = [k for k in keys_a if "Date" not in k]
+            keys_b = [k for k in keys_b if "Date" not in k]
+        else:
+            raise ValueError(
+                f"exclude is {exclude} but must be GLOBAL, DATE, or ATTR")
+
+    # check for equal length
+    if len(keys_a) != len(keys_b):
+        is_identical = _compare_sofa_warning((
+            f"not identical: sofa_a has {len(keys_a)} attributes for "
+            f"comparison and sofa_b has {len(keys_b)}."), verbose)
+
+        return is_identical
+
+    # check if the keys match
+    if set(keys_a) != set(keys_b):
+        is_identical = _compare_sofa_warning(
+            "not identical: sofa_a and sofa_b do not have the ame attributes",
+            verbose)
+
+        return is_identical
+
+    # compare the data inside the SOFA object
+    for key in keys_a:
+
+        # get data and types
+        a = getattr(sofa_a, key)
+        b = getattr(sofa_b, key)
+        type_a = sofa_a._convention[key]["type"]
+        type_b = sofa_b._convention[key]["type"]
+
+        # compare attributes
+        if type_a == "attribute" and type_b == "attribute":
+
+            # handling versions (might be integer, float, or string)
+            if not isinstance(a, str) or not isinstance(a, str):
+                a = str(float(a))
+                b = str(float(b))
+
+            # compare
+            if a != b:
+                is_identical = _compare_sofa_warning(
+                    f"not identical: different values for {key}", verbose)
+
+        # compare double variables
+        elif type_a == "double" and type_b == "double":
+
+            try:
+                npt.assert_allclose(np.squeeze(a), np.squeeze(b))
+            except AssertionError:
+                is_identical = _compare_sofa_warning(
+                    "not identical: different values for {key}", verbose)
+
+        # compare string variables
+        elif type_a == "string" and type_b == "string":
+            try:
+                assert np.all(
+                    np.squeeze(a).astype("S") == np.squeeze(b).astype("S"))
+            except AssertionError:
+                is_identical = _compare_sofa_warning(
+                    "not identical: different values for {key}", verbose)
+        else:
+            is_identical = _compare_sofa_warning(
+                (f"not identical: {key} has different data types "
+                 f"({type_a}, {type_b})"), verbose)
+
+    return is_identical
+
+
+def _compare_sofa_warning(message, verbose):
+    if verbose:
+        warnings.warn(message)
+    return False
```

Comment:
What's the intention of this? It doesn't really compare anything

Reply:
It is used to raise warnings by `equals` (previously `compare_sofa`) . renamed it to _equals_raise_warning